main/src/ca/uwaterloo/flix/language/phase/njvm/Mnemonics.scala

```diff
@@ -167,6 +170,72 @@ object Mnemonics {
       this.asInstanceOf[F[S]]
     }
 
+    /**
+      * Emits a getStatic instruction given the fieldName and it's type
+      *
+      * @param fieldName the of the field
+      * @param fieldType the jvmType of the field we want to get
+      * @pre This method should only be called if field with the specified fieldName matches the jvmType,
+      *      to avoid verifier errors
+      */
+    def emitGetStatic[S](fieldName: String, fieldType: NJvmType): F[S] = {
+      mv.visitFieldInsn(GETSTATIC, ct.name.toInternalName, fieldName, fieldType.toDescriptor)
+      this.asInstanceOf[F[S]]
+    }
+
+    /**
+      * Emits a getStatic instruction given the fieldName and it's type
+      *
+      * @param fieldName the of the field
+      * @param fieldType the jvmType of the field we want to get
+      * @pre This method should only be called if field with the specified fieldName matches the jvmType,
+      *      to avoid verifier errors
+      */
+    def emitPutStatic[S](fieldName: String, fieldType: NJvmType): F[S] = {
+      mv.visitFieldInsn(PUTSTATIC, ct.name.toInternalName, fieldName, fieldType.toDescriptor)
+      this.asInstanceOf[F[S]]
+    }
+
+
+
+
+    /**
+      * Emits a getField instruction given the fieldName and it's type
+      *
+      * @param fieldName the of the field
+      * @param fieldType the jvmType of the field we want to get
+      * @pre This method should only be called if field with the specified fieldName matches the jvmType,
+      *      to avoid verifier errors
+      */
+    def emitGetBoxedField[S](fieldName: String, fieldType: NJvmType): F[S] = {
```

Comment:
I find the name confusing. Is the field boxed or not? Is the result boxed? Maybe we should call it getFieldAndBox or similar